lib/mayaUsd/ufe/private/Utils.cpp

```diff
@@ -21,13 +21,37 @@
 #include <pxr/usd/usd/primCompositionQuery.h>
 #include <pxr/usd/usdGeom/xformable.h>
 
+#include <maya/MGlobal.h>
 #include <ufe/log.h>
 
 #include <memory>
 #include <string>
 
 PXR_NAMESPACE_USING_DIRECTIVE
 
+namespace {
+// find position index for a given layer in the local LayerStack ( strong-to-weak order )
+uint32_t findLayerIndex(const PXR_NS::UsdAttribute& attr, const PXR_NS::SdfLayerHandle& layer)
+{
+    const auto& prim = attr.GetPrim();
+    const auto& stage = prim.GetStage();
+    const auto& layerStack = stage->GetLayerStack();
```

Comment:
This is all fine, of course --- but I wonder performance-wise.  Unless the layer stack changes (and I think there are USD events to detect this --- maybe this one
https://graphics.pixar.com/usd/docs/api/class_sdf_notice_1_1_layers_did_change.html
but the docs appear broken), you can compute the index once and cache it.  Maybe we can wait for this optimization until we figure out what Pixar will be giving us (or when they will give it to us :) ).

I can see something like
std::unordered_map<SdfLayerHandle, int> layerIndices;

Reply:
@ppt-adsk caching the index compute should be trivial. We can add it at any-times if performance becomes a bottleneck. 

Honestly, how many layers are we talking about? Even with 500s layers which is a lot of layers to me we should have a very fast look up. 

My hope with Pixar is that the client(s) of such APIs shouldn't even think about such details. It should all happen behind the scene.

Reply:
I'm less concerned about the number of layers than about the number of attributes to check...  Which might be hundreds of thousands.  But you're right, let's put that on the side for now, we can revisit this later.